quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/KustoSampleApp.java

```diff
@@ -0,0 +1,395 @@
+                if (!execute(kustoClient, databaseName, command, ExecutionType.CONTROL_COMMAND)) {
+                    die("Failed to create table or validate it exists");
+                }
+            }
+
+            /*
+                Learn More:
+                Kusto batches data for ingestion efficiency. The default batching policy ingests data when one of the following conditions are met:
+                 1) More than 1,000 files were queued for ingestion for the same table by the same user
+                 2) More than 1GB of data was queued for ingestion for the same table by the same user
+                 3) More than 5 minutes have passed since the first file was queued for ingestion for the same table by the same user
+                For more information about customizing the ingestion batching policy, see: https://docs.microsoft.com/azure/data-explorer/kusto/management/batchingpolicy
+
+                Tip 1: Though most users should be fine with the defaults, to speed up ingestion, such as during development and
+                 in this sample app, we opt to modify the default ingestion policy to ingest data after 10 seconds have passed.
+                Tip 2: This is generally a one-time configuration.
+                Tip 3: You can also skip the batching for some files using the Flush-Immediately property, though this option should be used with care as it is inefficient.
+             */
+            waitForUserToProceed(String.format("Alter the batching policy for '%s.%s'", databaseName, tableName));
+            String command = String.format(".alter table %s policy ingestionbatching @'%s'", tableName, BATCHING_POLICY);
+            if (!execute(kustoClient, databaseName, command, ExecutionType.CONTROL_COMMAND)) {
+                System.out.println("Failed to alter the ingestion policy, which could be the result of insufficient permissions. The sample will still run, though ingestion will be delayed for up to 5 minutes.");
+            }
+
+            // Learn More: For more information about Kusto Query Language (KQL), see: https://docs.microsoft.com/azure/data-explorer/write-queries
+            waitForUserToProceed(String.format("Get existing row count in '%s.%s'", databaseName, tableName));
+            execute(kustoClient, databaseName, String.format("%s | count", tableName), ExecutionType.QUERY);
+
+            for (Map<String, String> file : dataToIngest) {
+                IngestionProperties.DataFormat dataFormat = IngestionProperties.DataFormat.valueOf(file.get("format").toLowerCase());
+                boolean useExistingMapping = Boolean.parseBoolean(file.get("useExistingMapping"));
+                String mappingName = file.get("mappingName");
+
+                // Learn More: For more information about providing inline mappings and mapping references, see: https://docs.microsoft.com/azure/data-explorer/kusto/management/mappings
+                if (!useExistingMapping) {

```

Comment:
This code section makes me think we need OneClick to explicitly provide the mapping mode (None, Use exiting Mapping, Create Mapping, Use Mapping Value)

Reply:
I personally don't think this logic is complex, especially once we separate it to its own method (recall that `main` will be much simpler). Your call.